spec.md

```diff
@@ -1671,40 +1671,52 @@ by the following parameters.
 2.  A positive integer literal representing the number of elements in the
     memory.
 
-3.  A variable number of named ports, each being a read port, a write port, or
-    readwrite port.
-
-4.  A non-negative integer literal indicating the read latency, which is the
-    number of cycles after setting the port's read address before the
-    corresponding element's value can be read from the port's data field.
-
-5.  A positive integer literal indicating the write latency, which is the number
-    of cycles after setting the port's write address and data before the
-    corresponding element within the memory holds the new value.
-
-6.  A read-under-write flag indicating the behavior when a memory location is
+3.  A variable number of named ports, each having following parameters:
+    1.  A read flag indicating the read capability of this port.
+    2.  A write flag indicating the write capability of this port.
+    3.  If the port can read, a non-negative integer literal indicating the read
+        latency, which is the number of cycles after setting the port's read
+        address before the corresponding element's value can be read from the
+        port's rdata field.
+    4.  If the port can write, a positive integer literal indicating the write
+        latency, which is the number of cycles after setting the port's write
+        address and wdata before the corresponding element within the memory
+        holds the new value.
+4.  A read-under-write flag indicating the behavior when a memory location is
     written to while a read to that location is in progress.
+5.  An optional type representing the custom port of this memory. This custom
+    port is intended for post-synthesis flows, and should be ignored in
+    behavioral simulation.
 
 Integer literals for the number of elements and the read/write latencies _may
 not be string-encoded integer literals_.
 
 The following example demonstrates instantiating a memory containing 256 complex
 numbers, each with 16-bit signed integer fields for its real and imaginary
 components. It has two read ports, `r1`{.firrtl} and `r2`{.firrtl}, and one
-write port, `w`{.firrtl}. It is combinationally read (read latency is zero
-cycles) and has a write latency of one cycle. Finally, its read-under-write
+write port, `w`{.firrtl}, with the ability to do partial writes using a mask.
+All of its read ports is combinationally read (read latency is zero cycles) and
+its write port has a write latency of one cycle. Finally, its read-under-write
 behavior is undefined.
 
 ``` firrtl
 mem mymem :
   data-type => {real:SInt<16>, imag:SInt<16>}
   depth => 256
-  reader => r1
-  reader => r2
-  writer => w
-  read-latency => 0
-  write-latency => 1
   read-under-write => undefined
+  port r1:
```

Comment:
It's weird to mix `=>` syntax and `:` syntax.

Reply:
The motivation of using `:` is to distinguish a indented block from a single configuration value. Using a different delimitator might make writing parser a little bit easier?